### PR TITLE
138 implementing optimistic UI

### PR DESCRIPTION
**Pull request by hunar1312**
Made adding notes UI follow Optimistic UI approach

### CODE REVIEW

File: app/components/assets/notes/notes.tsx [hunk 1]
```diff
@@ -1,18 +1,43 @@
...
 import type { NoteWithDate } from "./note";
 import { Note } from "./note";
 
 export const Notes = () => {
   const { asset } = useLoaderData();
+  const user = useUserData();
   const hasNotes = asset?.notes.length > 0;
-
+  const fetcher = useFetcher();
+  let onSubmissionContent = "";
```
Comment (DonKoko) [inline app/components/assets/notes/notes.tsx:16]: Can you add a comment here with what is going on. The code is not immediately obvious so I think its important for people who look at this code to understand it.

File: app/components/assets/notes/notes.tsx [hunk 2]
```diff
@@ -1,18 +1,43 @@
...
+  if (fetcher.formData) {
+    for (const data of fetcher.formData.entries()) {
+      onSubmissionContent = data[1].toString();
+    }
+  }
   return (
     <div>
-      <NewNote />
+      <div>{JSON.stringify(fetcher)}</div>
```
Comment (DonKoko) [inline app/components/assets/notes/notes.tsx]: This has to be cleaned up.

**Review by DonKoko — CHANGES_REQUESTED** [review]
Some changes are required. Overall great job.

File: app/components/assets/notes/notes.tsx (hunk 1 above)
Comment (hunar1312) [reply app/components/assets/notes/notes.tsx:16]: sure

File: app/components/assets/notes/notes.tsx (hunk 2 above)
Comment (hunar1312) [reply app/components/assets/notes/notes.tsx]: sorry for this one

**Review by DonKoko — APPROVED** [review]

**Review by DonKoko — APPROVED** [review]

**Pull request merged**